clients/spark-extensions-base/src/test/java/org/projectnessie/spark/extensions/AbstractSparkSqlTest.java

```diff
@@ -439,11 +440,50 @@ void useShowReferencesIn() throws NessieNotFoundException {
   }
 
   @Test
-  void useShowReferencesAtTimestamp() throws NessieNotFoundException, NessieConflictException {
+  void throwWhenUseShowReferencesAtTimestampWithoutTimeZone()
```

Comment:
> change the LocalDateTime.now(ZoneOffset.UTC) to something that does not use now(), but instead calculates it from the last commit to avoid the test to be prone to flakiness.

This particular test asserts that an exception is thrown when there is no time zone provided, so it does not matter if we pass `now()` or something else.


> have one commit before and one commit after the one the test asserts against. This would add a good amount of coverage to NessieUtils.findReferenceFromTimestamp

AND 

>add a test providing a timestamp that's before the first commit (throws an exception)

Done